packages/shortest/src/core/runner/test-reporter.ts

```diff
@@ -1,112 +1,126 @@
 import pc from "picocolors";
-import { AssertionError, TestFunction } from "../../types";
-
-export type TestStatus = "pending" | "running" | "passed" | "failed";
-
-interface TokenMetrics {
-  input: number;
-  output: number;
-}
-
-interface TestResult {
-  name: string;
-  status: TestStatus;
-  error?: Error;
-  tokenUsage?: TokenMetrics;
-}
+import { FileResult, TestResult, TestStatus } from "@/core/runner/index";
+import { AssertionError, TestFunction } from "@/types/test";
 
 export class TestReporter {
   private currentFile: string = "";
   private testResults: Record<string, TestResult> = {};
   private startTime: number = Date.now();
   private currentTest: TestResult | null = null;
+  private legacyOutputEnabled: boolean;
+  // private reporterLog: Log;
 
   // token pricing (Claude 3.5 Sonnet)
   private readonly COST_PER_1K_INPUT_TOKENS = 0.003;
   private readonly COST_PER_1K_OUTPUT_TOKENS = 0.015;
 
-  initializeTest(test: TestFunction) {
-    const testKey = test.name || "Unnamed Test";
-    this.currentTest = {
-      name: testKey,
-      status: "pending",
-    };
-    this.testResults[testKey] = this.currentTest;
+  private filesCount: number = 0;
```

Comment:
Simplified the logic a bit, and got rid of `this.testResults` & co. May bring similar functionality back in the future if needed.